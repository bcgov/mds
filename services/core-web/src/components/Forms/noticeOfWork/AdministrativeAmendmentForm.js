import React from "react";
import PropTypes from "prop-types";
import { Field, reduxForm, getFormValues } from "redux-form";
import { connect } from "react-redux";
import { compose } from "redux";
import { Form } from "@ant-design/compatible";
import "@ant-design/compatible/assets/index.css";
import { Button, Col, Row, Popconfirm } from "antd";
import { required, dateNotInFuture } from "@common/utils/Validate";
import CustomPropTypes from "@/customPropTypes";
import { resetForm, createDropDownList, formatDate } from "@common/utils/helpers";
import * as FORM from "@/constants/forms";
import { renderConfig } from "@/components/common/config";
import { getPermits } from "@common/selectors/permitSelectors";
import { getAmendmentReasonCodeDropdownOptions } from "@common/selectors/staticContentSelectors";

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
  formValues: PropTypes.objectOf(PropTypes.any),
  permits: PropTypes.arrayOf(CustomPropTypes.permit).isRequired,
  amendmentReasonCodeOptions: CustomPropTypes.options.isRequired,
};

const defaultProps = {
  formValues: {},
};

export const AdministrativeAmendmentForm = (props) => {
  const permitDropdown = createDropDownList(props.permits, "permit_no", "permit_id");
  const permitAmendments = props.permits.filter(
    ({ permit_id }) => permit_id === props.formValues.permit_id
  )[0];

  const amendmentDropdown = props.formValues.permit_id
    ? createDropDownList(
        permitAmendments.permit_amendments.filter(
          ({ permit_amendment_status_code }) => permit_amendment_status_code !== "DFT"
        ),
        "issue_date",
        "permit_amendment_guid",
        false,
        null,
        formatDate
      )
    : [];

  return (
    <Form layout="vertical" onSubmit={props.handleSubmit}>
      <Row>
        <Col span={24}>
          <Form.Item>
            <Field
              id="permit_id"
              name="permit_id"
              placeholder="Select a Permit"
              label="Permit*"
              component={renderConfig.SELECT}
              data={permitDropdown}
              validate={[required]}
            />
          </Form.Item>
          {props.formValues.permit_id && (
            <Form.Item>
              <Field
                id="permit_amendment_guid"
                name="permit_amendment_guid"
                label="Source Amendment by Issue Date*"
                component={renderConfig.SELECT}
                data={amendmentDropdown}
                validate={[required]}
              />
            </Form.Item>
          )}
          <Form.Item>
            <Field
              id="amendment_reason_codes"
              name="amendment_reason_codes"
              label="Reason for Amendment*"
              component={renderConfig.MULTI_SELECT}
              data={props.amendmentReasonCodeOptions}
              validate={[required]}
            />
          </Form.Item>
          <Form.Item>
            <Field
              id="received_date"
              name="received_date"
              label="Received Date*"
              component={renderConfig.DATE}
              validate={[required, dateNotInFuture]}
            />
          </Form.Item>
        </Col>
      </Row>
      <div className="right center-mobile">
        <Popconfirm
          placement="topRight"
          title="Are you sure you want to cancel?"
          onConfirm={props.closeModal}
          okText="Yes"
          cancelText="No"
          disabled={props.submitting}
        >
          <Button className="full-mobile" type="secondary" disabled={props.submitting}>
            Cancel
          </Button>
        </Popconfirm>
        <Button className="full-mobile" type="primary" htmlType="submit" loading={props.submitting}>
          Proceed
        </Button>
      </div>
    </Form>
  );
};

AdministrativeAmendmentForm.propTypes = propTypes;
AdministrativeAmendmentForm.defaultProps = defaultProps;

export default compose(
  connect((state) => ({
    formValues: getFormValues(FORM.ADMINISTRATIVE_AMENDMENT_FORM)(state),
    permits: getPermits(state),
    amendmentReasonCodeOptions: getAmendmentReasonCodeDropdownOptions(state),
  })),
  reduxForm({
    form: FORM.ADMINISTRATIVE_AMENDMENT_FORM,
    touchOnBlur: false,
    onSubmitSuccess: resetForm(FORM.ADMINISTRATIVE_AMENDMENT_FORM),
    enableReinitialize: false,
  })
)(AdministrativeAmendmentForm);
