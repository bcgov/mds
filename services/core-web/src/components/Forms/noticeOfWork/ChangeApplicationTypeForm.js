import React from "react";
import { reduxForm, Field } from "redux-form";
import { Form } from "@ant-design/compatible";
import "@ant-design/compatible/assets/index.css";
import { Col, Row } from "antd";
import { resetForm } from "@common/utils/helpers";
import { required } from "@common/utils/Validate";
import * as FORM from "@/constants/forms";
import RenderSelect from "@/components/common/RenderSelect";

const propTypes = {};

export const ChangeApplicationTypeForm = () => (
  <Form layout="vertical">
    <Row gutter={16}>
      <Col span={24}>
        <Form.Item>
          <Field
            id="type_of_application"
            name="type_of_application"
            label="Application Type*"
            component={RenderSelect}
            data={[
              { value: "New Permit", label: "New Permit" },
              { value: "Amendment", label: "Amendment" },
            ]}
            validate={[required]}
          />
        </Form.Item>
      </Col>
    </Row>
  </Form>
);

ChangeApplicationTypeForm.propTypes = propTypes;

export default reduxForm({
  form: FORM.CHANGE_NOW_TYPE,
  touchOnBlur: false,
  enableReinitialize: true,
  onSubmitSuccess: resetForm(FORM.CHANGE_NOW_TYPE),
})(ChangeApplicationTypeForm);
