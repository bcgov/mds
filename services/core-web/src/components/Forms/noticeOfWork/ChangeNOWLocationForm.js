import React, { Component } from "react";
import PropTypes from "prop-types";
import { compose } from "redux";
import { connect } from "react-redux";
import { reduxForm, Field, formValueSelector } from "redux-form";
import { Form, Button, Col, Row, Popconfirm, Badge } from "antd";
import { required, lat, lon } from "@common/utils/Validate";
import { resetForm } from "@common/utils/helpers";
import CustomPropTypes from "@/customPropTypes";
import * as FORM from "@/constants/forms";
import MineCard from "@/components/mine/NoticeOfWork/MineCard";
import RenderMineSelect from "@/components/common/RenderMineSelect";
import RenderField from "@/components/common/RenderField";
import * as Styles from "@/constants/styles";

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  locationOnly: PropTypes.bool,
  mine: CustomPropTypes.mine,
  latitude: PropTypes.string,
  longitude: PropTypes.string,
};

const defaultProps = {
  locationOnly: false,
  mine: {},
  latitude: "",
  longitude: "",
};

const selector = formValueSelector(FORM.CHANGE_NOW_LOCATION);
// eslint-disable-next-line react/prefer-stateless-function
export class ChangeNOWLocationForm extends Component {
  state = { submitting: false };
  handleFormSubmit = (values) => {
    this.setState({ submitting: true });
    this.props.handleSubmit(values);
  };
  render() {
    const additionalPin =
      this.props.latitude && this.props.longitude
        ? [this.props.latitude, this.props.longitude]
        : [];
    const span = this.props.locationOnly ? 12 : 6;
    return (
      <Form layout="vertical" onSubmit={this.handleFormSubmit}>
        <Row gutter={16}>
          {!this.props.locationOnly && (
            <Col md={12} s={24}>
              <Form.Item>
                <Field
                  id="mine_guid"
                  name="mine_guid"
                  component={RenderMineSelect}
                  validate={[required]}
                  showCard
                  fullWidth
                  additionalPin={additionalPin}
                />
              </Form.Item>
            </Col>
          )}
          <Col md={span} s={12}>
            <Form.Item label={[<Badge color={Styles.COLOR.yellow} />, "NoW Latitude"]}>
              <Field id="latitude" name="latitude" component={RenderField} validate={[lat]} />
            </Form.Item>
          </Col>
          <Col md={span} s={12}>
            <Form.Item label="NoW Longitude">
              <Field id="longitude" name="longitude" component={RenderField} validate={[lon]} />
            </Form.Item>
          </Col>
        </Row>
        {this.props.locationOnly && (
          <MineCard mine={this.props.mine} additionalPin={additionalPin} />
        )}
        <div className="right center-mobile">
          {this.props.locationOnly && (
            <Popconfirm
              placement="topRight"
              title="Are you sure you want to cancel?"
              onConfirm={this.props.closeModal}
              okText="Yes"
              cancelText="No"
            >
              <Button className="full-mobile" type="secondary">
                Cancel
              </Button>
            </Popconfirm>
          )}
          <Button
            className="full-mobile"
            type="primary"
            htmlType="submit"
            disabled={this.state.submitting}
          >
            {this.props.title}
          </Button>
        </div>
      </Form>
    );
  }
}

ChangeNOWLocationForm.propTypes = propTypes;
ChangeNOWLocationForm.defaultProps = defaultProps;

export default compose(
  connect((state) => ({
    latitude: selector(state, "latitude"),
    longitude: selector(state, "longitude"),
  })),
  reduxForm({
    form: FORM.CHANGE_NOW_LOCATION,
    onSubmitSuccess: resetForm(FORM.CHANGE_NOW_LOCATION),
  })
)(ChangeNOWLocationForm);
