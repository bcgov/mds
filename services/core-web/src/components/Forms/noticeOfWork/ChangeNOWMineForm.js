import React from "react";
import PropTypes from "prop-types";
import { reduxForm, Field } from "redux-form";
import { Form, Button, Col, Row, Popconfirm } from "antd";
import { resetForm } from "@common/utils/helpers";
import { required } from "@common/utils/Validate";
import * as FORM from "@/constants/forms";
import RenderMineSelect from "@/components/common/RenderMineSelect";

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  submitting: PropTypes.bool.isRequired,
};

export const ChangeNOWMineForm = (props) => (
  <Form layout="vertical" onSubmit={props.handleSubmit}>
    <Row gutter={16}>
      <Col>
        <Form.Item>
          <Field
            id="mine_guid"
            name="mine_guid"
            component={RenderMineSelect}
            validate={[required]}
            showCard
          />
        </Form.Item>
      </Col>
    </Row>
    <div className="right center-mobile">
      <Popconfirm
        placement="topRight"
        title="Are you sure you want to cancel?"
        onConfirm={props.closeModal}
        okText="Yes"
        cancelText="No"
      >
        <Button className="full-mobile" type="secondary">
          Cancel
        </Button>
      </Popconfirm>
      <Button className="full-mobile" type="primary" htmlType="submit" disabled={props.submitting}>
        {props.title}
      </Button>
    </div>
  </Form>
);

ChangeNOWMineForm.propTypes = propTypes;

export default reduxForm({
  form: FORM.CHANGE_NOW_MINE,
  touchOnBlur: false,
  onSubmitSuccess: resetForm(FORM.CHANGE_NOW_MINE),
})(ChangeNOWMineForm);
