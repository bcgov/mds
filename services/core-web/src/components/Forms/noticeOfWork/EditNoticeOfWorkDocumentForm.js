import React, { Component } from "react";
import { connect } from "react-redux";
import { compose } from "redux";
import PropTypes from "prop-types";
import { Field, reduxForm } from "redux-form";
import { Form } from "@ant-design/compatible";
import "@ant-design/compatible/assets/index.css";
import { Button, Col, Row, Popconfirm } from "antd";
import { required, maxLength, validateSelectOptions } from "@common/utils/Validate";
import { resetForm } from "@common/utils/helpers";
import { getDropdownNoticeOfWorkApplicationDocumentTypeOptions } from "@common/selectors/staticContentSelectors";
import { NOTICE_OF_WORK_DOCUMENT } from "@common/constants/API";
import { renderConfig } from "@/components/common/config";
import * as FORM from "@/constants/forms";
import CustomPropTypes from "@/customPropTypes";
import FileUpload from "@/components/common/FileUpload";

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  now_application_guid: PropTypes.string.isRequired,
  submitting: PropTypes.bool.isRequired,
  dropdownNoticeOfWorkApplicationDocumentTypeOptions: PropTypes.arrayOf(
    CustomPropTypes.dropdownListItem
  ).isRequired,
  change: PropTypes.func.isRequired,
  categoriesToShow: PropTypes.arrayOf(PropTypes.string),
};

const defaultProps = {
  categoriesToShow: [],
};

export class EditNoticeOfWorkDocumentForm extends Component {
  state = {
    uploadedFiles: [],
    disabled: false,
  };

  toggleDisabled = (value) => {
    this.setState({
      disabled: value,
    });
  };

  onFileLoad = (documentName, document_manager_guid) => {
    this.setState(
      (prevState) => ({
        uploadedFiles: [[document_manager_guid, documentName], ...prevState.uploadedFiles],
      }),
      () => this.toggleDisabled(false)
    );
    this.props.change("uploadedFiles", this.state.uploadedFiles);
  };

  onRemoveFile = (err, fileItem) => {
    this.setState((prevState) => ({
      uploadedFiles: prevState.uploadedFiles.filter((fileArr) => fileArr[0] !== fileItem.serverId),
    }));

    this.props.change("uploadedFiles", this.state.uploadedFiles);
  };

  render() {
    const filteredDropDownOptions = this.props.dropdownNoticeOfWorkApplicationDocumentTypeOptions.filter(
      ({ value }) =>
        this.props.categoriesToShow.length > 0 ? this.props.categoriesToShow.includes(value) : value
    );

    return (
      <Form layout="vertical" onSubmit={this.props.handleSubmit}>
        <Row gutter={16}>
          <Col span={24}>
            <Form.Item>
              <Field
                id="now_application_document_type_code"
                name="now_application_document_type_code"
                label="Document Category*"
                placeholder="Select a document type"
                component={renderConfig.SELECT}
                data={filteredDropDownOptions}
                validate={[
                  required,
                  validateSelectOptions(
                    this.props.dropdownNoticeOfWorkApplicationDocumentTypeOptions
                  ),
                ]}
              />
            </Form.Item>
            <Form.Item>
              <Field
                id="description"
                name="description"
                label="Description"
                component={renderConfig.AUTO_SIZE_FIELD}
                validate={[maxLength(280)]}
              />
            </Form.Item>
            <Form.Item>
              <Field
                id="is_final_package"
                name="is_final_package"
                label="Part of final application package"
                type="checkbox"
                component={renderConfig.CHECKBOX}
              />
            </Form.Item>
            <h5>Document Upload*</h5>
            <p className="p-light">
              All files uploaded will be classified using the selected Category. To upload other
              file types, re-open this form after submitting the current files.
            </p>
            <br />
            <Form.Item>
              <Form.Item>
                <Field
                  id="notice_of_work_file_upload"
                  name="notice_of_work_file_upload"
                  onFileLoad={this.onFileLoad}
                  onRemoveFile={this.onRemoveFile}
                  component={FileUpload}
                  addFileStart={() => this.toggleDisabled(true)}
                  uploadUrl={NOTICE_OF_WORK_DOCUMENT(this.props.now_application_guid)}
                  allowMultiple
                  allowRevert
                />
              </Form.Item>
            </Form.Item>
          </Col>
        </Row>
        <div className="right center-mobile">
          <Popconfirm
            placement="topRight"
            title="Are you sure you want to cancel?"
            onConfirm={this.props.closeModal}
            okText="Yes"
            cancelText="No"
          >
            <Button className="full-mobile" type="secondary">
              Cancel
            </Button>
          </Popconfirm>
          <Button
            className="full-mobile"
            type="primary"
            htmlType="submit"
            loading={this.props.submitting}
            disabled={
              this.state.uploadedFiles.length === 0 || this.props.submitting || this.state.disabled
            }
          >
            {this.props.title}
          </Button>
        </div>
      </Form>
    );
  }
}

EditNoticeOfWorkDocumentForm.propTypes = propTypes;
EditNoticeOfWorkDocumentForm.defaultProps = defaultProps;

export default compose(
  connect((state) => ({
    dropdownNoticeOfWorkApplicationDocumentTypeOptions: getDropdownNoticeOfWorkApplicationDocumentTypeOptions(
      state
    ),
  })),
  reduxForm({
    form: FORM.EDIT_NOTICE_OF_WORK_DOCUMENT_FORM,
    touchOnBlur: false,
    onSubmitSuccess: resetForm(FORM.EDIT_NOTICE_OF_WORK_DOCUMENT_FORM),
  })
)(EditNoticeOfWorkDocumentForm);
