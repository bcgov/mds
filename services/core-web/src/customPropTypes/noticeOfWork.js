import { PropTypes, shape } from "prop-types";

// This file is anticipated to have multiple exports
// eslint-disable-next-line import/prefer-default-export
export const nowApplication = shape({
  application_guid: PropTypes.string.isRequired,
  mine_guid: PropTypes.string,
  mine_name: PropTypes.string,
  minenumber: PropTypes.string,
  trackingnumber: PropTypes.number,
  applicationtype: PropTypes.string,
  status: PropTypes.string,
  submitteddate: PropTypes.string,
  receiveddate: PropTypes.string,
  applicantclientid: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  submitterclientid: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  noticeofworktype: PropTypes.string,
  typeofpermit: PropTypes.string,
  typeofapplication: PropTypes.string,
  latitude: PropTypes.string,
  longitude: PropTypes.string,
  nameofproperty: PropTypes.string,
  tenurenumbers: PropTypes.string,
  crowngrantlotnumbers: PropTypes.string,
  sitedirections: PropTypes.string,
  firstaidequipmentonsite: PropTypes.string,
  firstaidcertlevel: PropTypes.string,
  descexplorationprogram: PropTypes.string,
  proposedstartdate: PropTypes.string,
  proposedenddate: PropTypes.string,
  yearroundseasonal: PropTypes.string,
  landcommunitywatershed: PropTypes.string,
  landprivate: PropTypes.string,
  landlegaldesc: PropTypes.string,
  archsitesaffected: PropTypes.string,
  sandgravelquarryoperations: PropTypes.string,
  storeexplosivesonsite: PropTypes.string,
  bcexplosivespermitissued: PropTypes.string,
  bcexplosivespermitnumber: PropTypes.string,
  bcexplosivespermitexpiry: PropTypes.string,
  campdisturbedarea: PropTypes.number,
  camptimbervolume: PropTypes.string,
  bldgdisturbedarea: PropTypes.number,
  bldgtimbervolume: PropTypes.string,
  stgedisturbedarea: PropTypes.number,
  stgetimbervolume: PropTypes.string,
  fuellubstoreonsite: PropTypes.string,
  fuellubstored: PropTypes.number,
  fuellubstoremethodbulk: PropTypes.string,
  fuellubstoremethodbarrel: PropTypes.string,
  cbsfreclamation: PropTypes.string,
  cbsfreclamationcost: PropTypes.number,
  mechtrenchingreclamation: PropTypes.string,
  mechtrenchingreclamationcost: PropTypes.number,
  expsurfacedrillreclamation: PropTypes.string,
  expsurfacedrillreclcorestorage: PropTypes.string,
  expsurfacedrillreclamationcost: PropTypes.number,
  expaccessreclamation: PropTypes.string,
  expaccessreclamationcost: PropTypes.number,
  surfacebulksampleprocmethods: PropTypes.string,
  surfacebulksamplereclamation: PropTypes.string,
  surfacebulksamplereclsephandl: PropTypes.string,
  surfacebulksamplerecldrainmiti: PropTypes.string,
  surfacebulksamplereclcost: PropTypes.number,
  underexptotalore: PropTypes.string,
  underexptotaloreunits: PropTypes.string,
  underexptotalwaste: PropTypes.string,
  underexptotalwasteunits: PropTypes.string,
  underexpreclamation: PropTypes.string,
  underexpreclamationcost: PropTypes.number,
  placerundergroundoperations: PropTypes.string,
  placerhandoperations: PropTypes.string,
  placerreclamationarea: PropTypes.number,
  placerreclamation: PropTypes.string,
  placerreclamationcost: PropTypes.number,
  sandgrvqrydepthoverburden: PropTypes.string,
  sandgrvqrydepthtopsoil: PropTypes.string,
  sandgrvqrystabilizemeasures: PropTypes.string,
  sandgrvqrywithinaglandres: PropTypes.string,
  sandgrvqryalrpermitnumber: PropTypes.string,
  sandgrvqrylocalgovsoilrembylaw: PropTypes.string,
  sandgrvqryofficialcommplan: PropTypes.string,
  sandgrvqrylandusezoning: PropTypes.string,
  sandgrvqryendlanduse: PropTypes.string,
  sandgrvqrytotalmineres: PropTypes.string,
  sandgrvqrytotalmineresunits: PropTypes.string,
  sandgrvqryannualextrest: PropTypes.string,
  sandgrvqryannualextrestunits: PropTypes.string,
  sandgrvqryreclamation: PropTypes.string,
  sandgrvqryreclamationbackfill: PropTypes.string,
  sandgrvqryreclamationcost: PropTypes.number,
  sandgrvqrygrdwtravgdepth: PropTypes.string,
  sandgrvqrygrdwtrexistingareas: PropTypes.number,
  sandgrvqrygrdwtrtestpits: PropTypes.string,
  sandgrvqrygrdwtrtestwells: PropTypes.string,
  sandgrvqrygrdwtrother: PropTypes.string,
  sandgrvqrygrdwtrmeasprotect: PropTypes.string,
  sandgrvqryimpactdistres: PropTypes.string,
  sandgrvqryimpactdistwater: PropTypes.string,
  sandgrvqryimpactnoise: PropTypes.string,
  sandgrvqryimpactprvtaccess: PropTypes.string,
  sandgrvqryimpactprevtdust: PropTypes.string,
  sandgrvqryimpactminvisual: PropTypes.string,
  cutlinesexplgridtotallinekms: PropTypes.string,
  cutlinesexplgridtimbervolume: PropTypes.string,
  cutlinesreclamation: PropTypes.string,
  cutlinesreclamationcost: PropTypes.number,
  pondswastewatertreatfacility: PropTypes.string,
  freeusepermit: PropTypes.string,
  licencetocut: PropTypes.string,
  timbertotalvolume: PropTypes.string,
  campbuildstgetotaldistarea: PropTypes.number,
  mechtrenchingtotaldistarea: PropTypes.number,
  expsurfacedrilltotaldistarea: PropTypes.number,
  expaccesstotaldistarea: PropTypes.number,
  surfacebulksampletotaldistarea: PropTypes.number,
  placertotaldistarea: PropTypes.number,
  underexptotaldistarea: PropTypes.number,
  sandgrvqrytotaldistarea: PropTypes.number,
  pondstotaldistarea: PropTypes.number,
  reclcostsubtotal: PropTypes.number,
  reclcostexist: PropTypes.string,
  reclcostrecl: PropTypes.number,
  reclcosttotal: PropTypes.number,
  reclareasubtotal: PropTypes.number,
  reclareaexist: PropTypes.string,
  reclarearecl: PropTypes.number,
  reclareatotal: PropTypes.number,
  anyotherinformation: PropTypes.string,
  vfcbcapplicationurl: PropTypes.string,
  messagecreateddate: PropTypes.string,
  processed: PropTypes.string,
  processeddate: PropTypes.string,
  cutlinesexplgriddisturbedarea: PropTypes.number,
  pondsrecycled: PropTypes.string,
  pondsexfiltratedtoground: PropTypes.string,
  pondsdischargedtoenv: PropTypes.string,
  pondsreclamation: PropTypes.string,
  pondsreclamationcost: PropTypes.string,
  sandgrvqrytotalexistdistarea: PropTypes.number,
  nrsosapplicationid: PropTypes.string,
  isblastselect: PropTypes.string,
  istimberselect: PropTypes.string,
  applicant: PropTypes.objectOf(PropTypes.any),
  submitter: PropTypes.objectOf(PropTypes.any),
  documents: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  contacts: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  existing_placer_activity: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  existing_settling_pond: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  proposed_placer_activity: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  proposed_settling_pond: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  surface_bulk_sample_activity: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  sand_grv_qry_activity: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  under_exp_new_activity: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  under_exp_rehab_activity: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  under_exp_surface_activity: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  water_source_activity: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  exp_surface_drill_activity: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
});

export const activityDetails = shape({
  activity_type_description: PropTypes.string,
  disturbed_area: PropTypes.number,
  timber_volume: PropTypes.number,
  number_of_sites: PropTypes.number,
  width: PropTypes.number,
  length: PropTypes.number,
  depth: PropTypes.number,
  height: PropTypes.number,
  quantity: PropTypes.number,
  incline: PropTypes.number,
  incline_unit_type_code: PropTypes.string,
  cut_line_length: PropTypes.number,
  water_quantity: PropTypes.number,
  water_quantity_unit_type_code: PropTypes.string,
});

export const activityEquipment = shape({
  description: PropTypes.string,
  quantity: PropTypes.number,
  capacity: PropTypes.string,
});

export const activitySummary = shape({
  reclamation_description: PropTypes.string,
  reclamation_cost: PropTypes.number,
  total_disturbed_area: PropTypes.number,
  total_disturbed_area_unit_type_code: PropTypes.string,
  equipment: activityEquipment,
});

export const defaultActivity = shape({
  details: activityDetails,
  ...activitySummary,
});

export const camps = shape({
  camp_name: PropTypes.string,
  camp_number_people: PropTypes.number,
  camp_number_structures: PropTypes.number,
  has_fuel_stored: PropTypes.boolean,
  has_fuel_stored_in_bulk: PropTypes.boolean,
  has_fuel_stored_in_barrels: PropTypes.boolean,
  ...defaultActivity,
});

const surfaceDrilling = shape({
  reclamation_core_storage: PropTypes.string,
  ...defaultActivity,
});

const placer = shape({
  is_underground: PropTypes.boolean,
  is_hand_operation: PropTypes.boolean,
  reclamation_area: PropTypes.number,
  reclamation_unit_type_code: PropTypes.string,
  ...defaultActivity,
});

export const sandGravelQuarry = shape({
  average_overburden_depth: PropTypes.number,
  average_top_soil_depth: PropTypes.number,
  stability_measures_description: PropTypes.string,
  is_agricultural_land_reserve: PropTypes.boolean,
  agri_lnd_rsrv_permit_application_number: PropTypes.string,
  has_local_soil_removal_bylaw: PropTypes.boolean,
  community_plan: PropTypes.string,
  land_use_zoning: PropTypes.string,
  proposed_land_use: PropTypes.string,
  total_mineable_reserves: PropTypes.number,
  total_mineable_reserves_unit_type_code: PropTypes.string,
  total_annual_extraction: PropTypes.number,
  total_annual_extraction_unit_type_code: PropTypes.string,
  average_groundwater_depth: PropTypes.number,
  has_groundwater_from_existing_area: PropTypes.boolean,
  has_groundwater_from_test_pits: PropTypes.boolean,
  has_groundwater_from_test_wells: PropTypes.boolean,
  groundwater_from_other_description: PropTypes.string,
  groundwater_protection_plan: PropTypes.string,
  nearest_residence_distance: PropTypes.number,
  nearest_residence_distance_unit_type_code: PropTypes.string,
  nearest_water_source_distance: PropTypes.number,
  nearest_water_source_distance_unit_type_code: PropTypes.string,
  noise_impact_plan: PropTypes.string,
  secure_access_plan: PropTypes.string,
  dust_impact_plan: PropTypes.string,
  visual_impact_plan: PropTypes.string,
  reclamation_backfill_detail: PropTypes.string,
  ...defaultActivity,
});

export const settlingPond = shape({
  proponent_pond_name: PropTypes.string,
  is_ponds_exfiltrated: PropTypes.boolean,
  is_ponds_recycled: PropTypes.boolean,
  is_ponds_discharged: PropTypes.boolean,
  ...defaultActivity,
});

export const surfaceBulkSamples = shape({
  processing_method_description: PropTypes.string,
  handling_instructions: PropTypes.string,
  drainage_mitigation_description: PropTypes.string,
  ...defaultActivity,
});

export const waterSupply = shape({
  supply_source_description: PropTypes.string,
  supply_source_type: PropTypes.string,
  water_use_description: PropTypes.string,
  estimate_rate: PropTypes.number,
  pump_size: PropTypes.number,
  intake_location: PropTypes.string,
  ...defaultActivity,
});

export const statueOfLand = shape({
  has_community_water_shed: PropTypes.boolean,
  has_archaeology_sites_affected: PropTypes.boolean,
});

export const importedNOWApplication = shape({
  now_application_guid: PropTypes.string,
  mine_guid: PropTypes.string,
  mine_name: PropTypes.string,
  mine_no: PropTypes.string,
  mine_region: PropTypes.string,
  imported_to_core: PropTypes.boolean,
  notice_of_work_type_code: PropTypes.string,
  now_application_status_code: PropTypes.string,
  submitted_date: PropTypes.date,
  received_date: PropTypes.date,
  latitude: PropTypes.number,
  longitude: PropTypes.number,
  property_name: PropTypes.string,
  tenure_number: PropTypes.string,
  description_of_land: PropTypes.string,
  proposed_start_date: PropTypes.date,
  proposed_end_date: PropTypes.date,
  directions_to_site: PropTypes.string,

  contacts: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  submission_documents: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  blasting_operation: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  state_of_land: statueOfLand,
  camps,
  cut_lines_polarization_survey: defaultActivity,
  exploration_access: defaultActivity,
  exploration_surface_drilling: surfaceDrilling,
  mechanical_trenching: defaultActivity,
  sand_and_gravel: sandGravelQuarry,
  settling_pond: settlingPond,
  surface_bulk_sample: surfaceBulkSamples,
  underground_exploration: defaultActivity,
  water_supply: waterSupply,
  placer_operation: placer,
});
