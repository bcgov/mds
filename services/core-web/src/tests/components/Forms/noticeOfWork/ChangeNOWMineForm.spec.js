import React from "react";
import { shallow } from "enzyme";
import { ChangeNOWMineForm } from "@/components/Forms/noticeOfWork/ChangeNOWMineForm";
import { NOW } from "@/tests/mocks/dataMocks";

const dispatchProps = {};
const props = {};

const setupDispatchProps = () => {
  dispatchProps.handleSubmit = jest.fn();
  dispatchProps.handleChange = jest.fn();
  dispatchProps.handleSelect = jest.fn();
  dispatchProps.closeModal = jest.fn();
};

const setupProps = () => {
  props.title = "mockTitle";
  props.submitting = false;
  // eslint-disable-next-line prefer-destructuring
  props.noticeOfWork = NOW.applications[0];
};

beforeEach(() => {
  setupDispatchProps();
  setupProps();
});

describe("ChangeNOWMineForm", () => {
  it("renders properly", () => {
    const component = shallow(<ChangeNOWMineForm {...dispatchProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
});
