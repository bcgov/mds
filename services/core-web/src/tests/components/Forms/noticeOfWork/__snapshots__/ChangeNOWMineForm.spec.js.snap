// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeNOWMineForm renders properly 1`] = `
<Form
  colon={true}
  hideRequiredMark={false}
  layout="vertical"
  onSubmit={[MockFunction]}
>
  <Row
    gutter={16}
  >
    <Col>
      <FormItem
        hasFeedback={false}
      >
        <Field
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(RenderMineSelect)",
              "type": [Function],
            }
          }
          id="mine_guid"
          name="mine_guid"
          showCard={true}
          validate={
            Array [
              [Function],
            ]
          }
        />
      </FormItem>
    </Col>
  </Row>
  <div
    className="right center-mobile"
  >
    <Popconfirm
      cancelText="No"
      disabled={false}
      icon={
        <Icon
          theme="filled"
          type="exclamation-circle"
        />
      }
      okText="Yes"
      okType="primary"
      onConfirm={[MockFunction]}
      placement="topRight"
      title="Are you sure you want to cancel?"
      transitionName="zoom-big"
      trigger="click"
    >
      <Button
        block={false}
        className="full-mobile"
        ghost={false}
        htmlType="button"
        loading={false}
        type="secondary"
      >
        Cancel
      </Button>
    </Popconfirm>
    <Button
      block={false}
      className="full-mobile"
      disabled={false}
      ghost={false}
      htmlType="submit"
      loading={false}
      type="primary"
    >
      mockTitle
    </Button>
  </div>
</Form>
`;
